CONTRIBUTING.md

```diff
@@ -1,303 +1,303 @@
+├── tests         # Test cases for API / e2e testings
+└── tools         # Keep supporting tools
+...
+```
+
+The folder graph below shows the structure of the source code folder `harbor/src`, which will be your primary working directory. The key folders are also commented.
+```
+.
+├── adminserver          # Source code for the admin server component
+│   ├── api
+│   ├── auth
+│   ├── client
+│   ├── handlers
+│   ├── systemcfg
+│   └── systeminfo
+├── common              # Source code for some general components like dao etc.
+│   ├── api
+│   ├── config
+│   ├── dao
+│   ├── models
+│   ├── notifier
+│   ├── scheduler
+│   ├── secret
+│   ├── security
+│   └── utils
+├── jobservice          # Source code for the job service component
+│   ├── api
+│   ├── config
+│   ├── job
+│   ├── replication
+│   ├── scan
+│   └── utils
+├── ui                  # Source code for the harbor service component
+│   ├── api
+│   ├── auth
+│   ├── config
+│   ├── controllers
+│   ├── filter
+│   ├── promgr
+│   ├── proxy
+│   ├── service
+│   ├── static
+│   ├── utils
+│   └── views
+├── portal               # The code of harbor web UI
+│   ├── e2e
+│   ├── lib             # Source code of @harbor/ui npm library which includes the main UI components of web UI
+│   └── src             # General web page UI code of Harbor
+└── vendor              # Go code dependencies
+    ├── github.com
+    ├── golang.org
+    ├── google.golang.org
+    └── gopkg.in
+```
+
+### Setup Development Environment
+
+#### Go
+Harbor backend is written in [Go](http://golang.org/). If you don't have a Harbor backend service development environment, please [set one up](https://golang.org/doc/install).
+
+|  Harbor  |  Requires Go  |
+|----------|---------------|
+|   1.1    |    1.7.3      |
+|   1.2    |    1.7.3      |
+|   1.3    |    1.9.2      |
+|   1.4    |    1.9.2      |
+
+Ensure your GOPATH and PATH have been configured in accordance with the Go environment instructions.
+
+**Dependency Management:** Harbor uses [dep](https://github.com/golang/dep) for dependency management of go code.  The official maintainers will take the responsibility for managing the code in `vendor` directory.  Please don't try to submit a PR to update the dependency code, open an issue instead.  If your PR requires a change in the vendor code please make sure you discuss it with the maintainers in advance.
+
+#### Web
+
+Harbor web UI is built based on [Clarity](https://vmware.github.io/clarity/) and [Angular](https://angular.io/) web framework. To setup web UI development environment, please make sure the [npm](https://www.npmjs.com/get-npm) tool is installed first.
+
+|  Harbor  |  Requires Angular  |  Requires Clarity  |
+|----------|--------------------|--------------------|
+|   1.1    |      2.4.1         |       0.8.7        |
+|   1.2    |      4.1.3         |       0.9.8        |
+|   1.3    |      4.3.0         |       0.10.17      |
+|   1.4    |                    |                    |
```

Comment:
Same comments with above

Reply:
Ditto above with Angular / Clarity @zhoumeina

Reply:
@ninjadq 

Please help to provide the required info for @clouderati .

Reply:
| 1.4 | 4.3.0 | 0.10.17 |
| 1.5 | 4.3.0 | 0.10.27 |
| 1.6 | 4.3.0 | 0.10.27 |